packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx

```diff
@@ -140,26 +140,47 @@ export type AccessAndSaleFormValues = {
   [PREVIEW]?: number
 }
 
-const AccessAndSaleFormSchema = (trackLength: number) =>
-  z.object({
-    [PREMIUM_CONDITIONS]: z.nullable(
-      z.object({
-        // TODO: there are other types
-        usdc_purchase: z.object({
-          price: z
-            .number()
-            .lte(999, messages.errors.price.tooHigh)
-            .gte(99, messages.errors.price.tooLow)
+export const AccessAndSaleFormSchema = (trackLength: number) =>
+  z
+    .object({
+      [PREMIUM_CONDITIONS]: z.nullable(
+        z.object({
+          // TODO: there are other types
+          usdc_purchase: z.optional(
+            z.object({
+              price: z
+                .number()
+                .lte(999, messages.errors.price.tooHigh)
+                .gte(99, messages.errors.price.tooLow)
+            })
+          )
         })
-      })
-    ),
-    [PREVIEW]: z.optional(
-      z
-        .number()
-        .gte(0, messages.errors.preview.tooEarly)
-        .lte(trackLength - 15, messages.errors.preview.tooLate)
+      ),
+      [PREVIEW]: z.optional(z.number())
+    })
+    .refine(
+      (values) => {
+        const formValues = values as AccessAndSaleFormValues
+        if (isPremiumContentUSDCPurchaseGated(formValues[PREMIUM_CONDITIONS])) {
+          return formValues[PREVIEW] !== undefined && formValues[PREVIEW] >= 0
+        }
+        return true
+      },
+      { message: messages.errors.preview.tooEarly, path: [PREVIEW] }
+    )
+    .refine(
+      (values) => {
+        const formValues = values as AccessAndSaleFormValues
+        if (isPremiumContentUSDCPurchaseGated(formValues[PREMIUM_CONDITIONS])) {
+          return (
+            formValues[PREVIEW] === undefined ||
+            (formValues[PREVIEW] >= 0 && formValues[PREVIEW] < trackLength - 15)
```

Comment:
fwiw it turns out track previews are implemented on the content end as 30s, so maybe easiest to update the UI side? Either way, maybe a comment/constant for the 15 here?

Reply:
ah great call. I'll make it 30 before the end